arbos/tx_processor.go

```diff
@@ -481,15 +481,26 @@ func (p *TxProcessor) GasChargingHook(gasRemaining *uint64) (common.Address, mul
 	multiGas := multigas.L1CalldataGas(gasNeededToStartEVM)
 
 	if !p.msg.TxRunContext.IsEthcall() {
-		// If this is a real tx, limit the amount of computed based on the gas pool.
-		// We do this by charging extra gas, and then refunding it later.
-		gasAvailable, _ := p.state.L2PricingState().PerBlockGasLimit()
-		if *gasRemaining > gasAvailable {
-			p.computeHoldGas = *gasRemaining - gasAvailable
-			*gasRemaining = gasAvailable
-			// The amount of multigas does not increase here because it will be refunded later.
+		// Before ArbOS 50, use the block gas limit to limit the transaction gas.
+		max, err := p.state.L2PricingState().PerBlockGasLimit()
```

Comment:
Minor tweak, but it'd probably make sense to put this in an else by the version to not have the unnecessary state read here

Reply:
Done.